Link QNN backend to pybinding lib when built

GregoryComer opened the pull request:
This change makes it possible to run QNN models with pybindings, when enabled. It does still require the environment setup described in the docs.

I validated this change locally by running install_executorch.sh with CMAKE_ARGS="-DEXECUTORCH_BUILD_QNN=ON", and then lowering and running a simple add model. The CI set up further in this stack also leverage this feature and test it extensively.

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer commented:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* (to be filled)

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

digantdesai reviewed (commented):
I guess rest of the pybind code was already present?

GregoryComer commented:
> I guess rest of the pybind code was already present?

When all of the required libraries and environment variables are set up, it will work with only this change. I would like to further simplify, as there is a lot of setup required - including some copying around of python extension binaries. But this code here is enough to make it runnable in conjunction with the existing QNN CI scripts (which do all of the setup).

GregoryComer commented:
> > as there is a lot of setup required - including some copying around of python extension binaries
> 
> Can you elaborate?

Yeah, it looks like it's currently building the Python extension components separately and then copying them into the ET directory. (See https://docs.pytorch.org/executorch/main/backends-qualcomm.html#aot-ahead-of-time-components). As part of the build cleanup and pybind enablement, I'll see if I can fold that into our "main" pybind build. That will also make it just work with installation, wheels, etc.

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

cccclai commented:
> > > as there is a lot of setup required - including some copying around of python extension binaries
> > 
> > 
> > Can you elaborate?
> 
> Yeah, it looks like it's currently building the Python extension components separately and then copying them into the ET directory. (See https://docs.pytorch.org/executorch/main/backends-qualcomm.html#aot-ahead-of-time-components). As part of the build cleanup and pybind enablement, I'll see if I can fold that into our "main" pybind build. That will also make it just work with installation, wheels, etc.

They should be part of `./backends/qualcomm/scripts/build.sh`. After you run this script, does pybind work?

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
[Backend Tester] Add QNN test flow to nightly CI (#13468)

Add (unquantized) QNN test flows to the nightly CI. This runs both the
model and operator suite.

GregoryComer committed:
Update

[ghstack-poisoned]

GregoryComer committed:
Update

[ghstack-poisoned]

cccclai reviewed (approved):
Thanks for enabling pybinding + QNN
